src/bun.js/webcore/FileReader.zig

```diff
@@ -613,13 +613,7 @@ pub fn onReaderDone(this: *FileReader) void {
                         globalThis,
                         .js_undefined,
                         &.{
-                            JSC.ArrayBuffer.fromBytes(
-                                buffered.items,
-                                .Uint8Array,
-                            ).toJS(
-                                globalThis,
-                                null,
-                            ),
+                            JSC.ArrayBuffer.fromBytes(buffered.items, .Uint8Array).toJS(globalThis, null) catch .zero, // TODO: properly propagate exception upwards
                         },
```

Comment:
```suggestion
                        &.{JSC.ArrayBuffer.fromBytes(buffered.items, .Uint8Array).toJS(globalThis, null) catch |err| {
                            this.pending.result = .{ .err = .{ .WeakJSValue = globalThis.takeException(err) } };
                            return;
                        }},
```

or add another union field for JSError that takes the exception when rejecting the promise